bundle/jsky.app.ot/src/test/scala/jsky/app/ot/gemini/editor/sitequality/TimingWindowImporterSpec.scala

```diff
@@ -0,0 +1,110 @@
+package jsky.app.ot.gemini.editor.sitequality
+
+import java.time.Instant
+
+import edu.gemini.spModel.gemini.obscomp.SPSiteQuality.TimingWindow
+import jsky.app.ot.gemini.editor.sitequality.TimingWindowParser.{TimingWindowParseFailure, TimingWindowParseResults}
+import org.specs2.ScalaCheck
+import org.specs2.mutable.Specification
+
+import scalaz._
+import Scalaz._
+
+object TimingWindowImporterSpec extends Specification with ScalaCheck {
+  "timing window parsing" should {
+    "handle empty inputs" in {
+      val results = TimingWindowParser.parseTimingWindows(Nil)
+      (results.successes.isEmpty must beTrue) and (results.failures.isEmpty must beTrue)
+    }
+
+    "ignore empty lines and comments" in {
+      val lines = List("", "  ", "#", "# Comment", " # abc ", "    ## nothing here")
+      val results = TimingWindowParser.parseTimingWindows(lines)
+      (results.successes.isEmpty must beTrue) and (results.failures.isEmpty must beTrue)
+    }
+
+    "create timing windows from complete data, ignoring whitespace" in {
+      val t1Tup = ("2018-01-01", "01:02:03", "04:05", 1.some, "01:02:03".some)
+      val t2Tup = ("   2018-01-02  ", "  02:03:04   ", "   06:07  ", 2.some, "   03:04:05    ".some)
+      val (successes, _, results) = runParsing(t1Tup.right, t2Tup.right)
+      (successes must === (results.successes)) and (results.failures.isEmpty must beTrue)
+    }
+
+    "identify a success and a failure" in {
+      val t1Tup = ("2017-06-10", "23:59:59", "48:00", 3.some, "123:45:00".some)
+      val fail1 = "Ceci n'est pas un timing window."
```

Comment:
I was just trying to appear cultured since I so rarely have the opportunity to do so! Haha.
![300px-magrittepipe](https://cloud.githubusercontent.com/assets/8795653/24256929/42577da0-0fc8-11e7-96c1-34d474091621.jpg)